src/Symfony/Component/FeatureToggle/Strategy/RequestHeaderStrategy.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureToggle\Strategy;
+
+use Symfony\Component\FeatureToggle\StrategyResult;
+
+// TODO: make it case insensitive ?
+final class RequestHeaderStrategy implements StrategyInterface
+{
+    public function __construct(
+        private readonly string $headerName,
+    ) {
+    }
+
+    public function compute(): StrategyResult
+    {
+        if (!array_key_exists('HTTP_'.$this->headerName, $_SERVER)) {
```

Comment:
This should not be based on `$_SERVER` but integration with the HttpFoundation component instead, which would make it work in all runtimes in which Symfony could be used.

Reply:
Agreed. But this is for standalone purposes. I was thinking of having both. One that is available in teh based component and the other one for the bundle. WDYT ?

Reply:
Having some strategies that make sense only for standalone usage but not for usage in the full stack framework (while they have similar needs) looks wrong to me. This would actually make it easier to write buggy code.
Even though we make components usable in a standalone way, we should still ensure that we provide the best DX possible when this gets used in the context of the framework.

Note that my previous message talked about runtimes as I was thinking of the runtime component to support more exotic deployments than php-fpm, but there is actually another case that is much more common that does not allow reading headers from super-globals due to using a special "runtime" (quotes are because this is not actually a runtime in the meaning of the runtime component): running functional tests with the WebTestCase.

Reply:
Ok. I didn't incldued yet because I wasn't sure on which Request I should rely on ? Current Request or Master request ? Both makes sense. I also thought of using both (if it differs).

Reply:
Ok I added new classes to base the strategy on RequestStack. ATM I didn't deleted the ones based on Super globals because I find it creates a heavy dependency when used outside the framework. Would that be okay to keep them or should I delete them anyway ?